web/lib/HabitatCommunities.js

```diff
@@ -1,61 +1,385 @@
+      </div>
+    </div>
+  </div>
+  <div slot='back'>
+    <div id='description-card'>
+      <div id='back-title'>
+        <a class='title'></a>
+      </div>
+        <div id='description'>
+          <p id='details'></p>
+        </div>
+    </div>
+  </div>
+</habitat-flip-card>
 `;
 
+class HabitatCommunities extends HabitatPanel {
+  
+  static TEMPLATE =
+  `
+  <style>
+  #communities, #user-communities {
+    gap: 2em;
+    margin-bottom: 5em;
+  }
+  .flip-card {
+    cursor: pointer;
+    min-width: 15em;
+    max-width: 30em;
+    flex: 1 1 0;
+  }
+  #buttons * {
+    margin-right: 0;
+  }
+  #create-community {
+    visibility:hidden;
+  }
+  #create-community.visible {
+    visibility:visible;
+  }
+  #create-community.active {
+    background: var(--color-bg-invert);
+    color: var(--color-text-invert);
+    transition: all .2s linear;
+  }
+  #sort {
+    display: block;
+    padding: .375em 1em;
+    margin-left: .5em;
+  }
+  #sort > svg > path {
+    fill: var(--color-text);
+  }
+  #sort-dropdown {
+    display:none;
+  }
+  #sort-dropdown.active {
+    display:block;
+  }
+  #sort-options {
+    position:absolute;
+    right:5%;
+    margin-top:2em;
+    z-index:1;
+    box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.5);
+    outline: none;
+  }
+  #sort-options * {
+    display:block;
+    padding:1em 0;
+    cursor:pointer;
+    color: var(--color-text);
+  }
+  #user-section {
+    display: none;
+  }
+  #user-section.visible {
+    display: block;
+  }
+
+  
+  </style>
+  <div style='width:90%;margin-left:auto;margin-right:auto;position:relative;'>
+
+    <space></space>
+
+    <div style='position:absolute;right:0;padding:1rem 0;'>
+      <div id='buttons' class='flex row' style='place-content:flex-end;'>
+        <button id='create-community'>Create</button>
+        <button id='sort'>${SVG_SORT_ICON}</button>
+        <div id='sort-dropdown' style='position:relative;'>
+          <ul class='box' id='sort-options'>
+            <li id='sort-az' >A - Z</li>
+            <li id='sort-za' >Z - A</li>
+            <li id='sort-recent' >Recent</li>
+          </ul>
+        </div>
+      </div>
+    </div>
+
+    <div id='creating-community'>
+    </div>
+
+    <div id='user-section'>
+      <p class='xl light' style='padding:2rem 0;'><span><emoji-herb></emoji-herb><span> Your Communities</span></span></p>
+      <div id='user-communities' class='flex row center evenly'></div>
+    </div>
+    <div>
+      <p class='xl light' style='padding:2rem 0;'><span><emoji-camping></emoji-camping><span> Communities on Habitat</span></span></p>
+      <div id='communities' class='flex row center evenly'></div>
+    </div>
+
+  </div>
+  `;
+
   constructor() {
     super();
 
+    this.tokens = []; //compare w each community token
+    //if user has a community's token, toggle once
+    this.userSection = this.shadowRoot.querySelector('#user-section');
+
+    // sorting arrays
+    this.communities = [];
+    this.userCommunities = [];
```

Comment:
this arrary muste be eventually cleared (along with any elements) if the user changes his wallet account for example